modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx

```diff
@@ -0,0 +1,174 @@
+	name: 'Object Name',
+	pluralLabel: {en_US: 'Plural Label Test'},
+	system: false,
+	titleObjectFieldName: '',
+} as Partial<ObjectDefinition>;
+
+describe('ObjectDataContainer component', () => {
+	it('check whether the object name input value is displayed correctly and is required', () => {
+		render(
+			<ObjectDataContainer
+				dbTableName=""
+				errors={{}}
+				handleChange={() => {}}
+				hasUpdateObjectDefinitionPermission={false}
+				isApproved={false}
+				isLinkedObjectDefinition={false}
+				onSubmit={() => {}}
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		const objectNameInput = document.getElementById(
+			'lfr-objects__object-data-container-name'
+		);
+
+		expect(objectNameInput).toBeRequired();
+		expect(objectNameInput).toHaveValue('Object Name');
+	});
+
+	it('check when object is published whether object name input is disabled', () => {
+		render(
+			<ObjectDataContainer
+				dbTableName=""
+				errors={{}}
+				handleChange={() => {}}
+				hasUpdateObjectDefinitionPermission={true}
+				isApproved={true}
+				isLinkedObjectDefinition={false}
+				onSubmit={() => {}}
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		const objectNameInput = document.getElementById(
+			'lfr-objects__object-data-container-name'
+		);
+
+		expect(objectNameInput).toBeDisabled();
+	});
+
+	it('check whether the object label input value is displayed correctly', () => {
+		render(
+			<ObjectDataContainer
+				dbTableName=""
+				errors={{}}
+				handleChange={() => {}}
+				hasUpdateObjectDefinitionPermission={false}
+				isApproved={false}
+				isLinkedObjectDefinition={false}
+				onSubmit={() => {}}
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		expect(screen.getByDisplayValue('Label Test')).toBeInTheDocument();
+	});
+
+	it('check whether the input value of the object plural label is displayed correctly', () => {
+		render(
+			<ObjectDataContainer
+				dbTableName=""
+				errors={{}}
+				handleChange={() => {}}
+				hasUpdateObjectDefinitionPermission={false}
+				isApproved={false}
+				isLinkedObjectDefinition={false}
+				onSubmit={() => {}}
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		expect(
+			screen.getByDisplayValue('Plural Label Test')
+		).toBeInTheDocument();
+	});
+
+	it('check if the object table name is displayed correctly and is disabled', () => {
+		render(
+			<ObjectDataContainer
+				dbTableName="DBTableName"
+				errors={{}}
+				handleChange={() => {}}
+				hasUpdateObjectDefinitionPermission={false}
+				isApproved={false}
+				isLinkedObjectDefinition={false}
+				onSubmit={() => {}}
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		const tableNameInput = screen.getByDisplayValue('DBTableName');
+
+		expect(tableNameInput).toBeInTheDocument();
+		expect(tableNameInput).toBeDisabled();
+	});
+
+	it('make sure active object toggle is not checked and disabled by default', () => {
+		render(
```

Comment:
To keep the tests result easier for reading, let's make simpler and more direct sentences, like:

```suggestion
	it('shows the toggle not checked and disabled by default', () => {
		render(
```